src/colorizer/IRenderCanvas.ts

```diff
@@ -1,77 +1,51 @@
-import { Vector2 } from "three";
-
-import { ViewerStoreState } from "@/state/slices";
-
-import { CanvasScaleInfo, FrameLoadResult, PixelIdInfo } from "./types";
-
-export type RenderCanvasStateParams = Pick<
-  ViewerStoreState,
-  | "dataset"
-  | "collection"
-  | "datasetKey"
-  | "pendingFrame"
-  | "featureKey"
-  | "track"
-  | "showTrackPath"
-  | "showTrackPathBreaks"
-  | "colorRamp"
-  | "colorRampRange"
-  | "categoricalPaletteRamp"
-  | "outlineColor"
-  | "edgeColor"
-  | "edgeColorAlpha"
-  | "edgeMode"
-  | "trackPathColor"
-  | "trackPathWidthPx"
-  | "trackPathColorMode"
-  | "outlierDrawSettings"
-  | "outOfRangeDrawSettings"
-  | "inRangeLUT"
-  | "vectorMotionDeltas"
-  | "vectorVisible"
-  | "vectorColor"
-  | "vectorScaleFactor"
-  | "backdropKey"
-  | "backdropVisible"
-  | "channelSettings"
-  | "objectOpacity"
-  | "backdropSaturation"
-  | "backdropBrightness"
->;
-
-export const renderCanvasStateParamsSelector = (state: ViewerStoreState): RenderCanvasStateParams => ({
-  dataset: state.dataset,
-  collection: state.collection,
-  datasetKey: state.datasetKey,
-  featureKey: state.featureKey,
-  pendingFrame: state.pendingFrame,
-  track: state.track,
-  showTrackPath: state.showTrackPath,
-  showTrackPathBreaks: state.showTrackPathBreaks,
-  colorRamp: state.colorRamp,
-  colorRampRange: state.colorRampRange,
-  categoricalPaletteRamp: state.categoricalPaletteRamp,
-  outlineColor: state.outlineColor,
-  edgeColor: state.edgeColor,
-  edgeMode: state.edgeMode,
-  edgeColorAlpha: state.edgeColorAlpha,
-  trackPathColor: state.trackPathColor,
-  trackPathWidthPx: state.trackPathWidthPx,
-  trackPathColorMode: state.trackPathColorMode,
-  outlierDrawSettings: state.outlierDrawSettings,
-  outOfRangeDrawSettings: state.outOfRangeDrawSettings,
-  inRangeLUT: state.inRangeLUT,
-  vectorMotionDeltas: state.vectorMotionDeltas,
-  vectorVisible: state.vectorVisible,
-  vectorColor: state.vectorColor,
-  vectorScaleFactor: state.vectorScaleFactor,
-  backdropKey: state.backdropKey,
-  backdropVisible: state.backdropVisible,
-  channelSettings: state.channelSettings,
-  objectOpacity: state.objectOpacity,
-  backdropSaturation: state.backdropSaturation,
-  backdropBrightness: state.backdropBrightness,
-});
+import { Color, Vector2 } from "three";
+
+import { ColorRamp, Dataset, Track } from "@/colorizer";
+import Collection from "@/colorizer/Collection";
+import {
+  CanvasScaleInfo,
+  ChannelSetting,
+  DrawMode,
+  DrawSettings,
+  FrameLoadResult,
+  PixelIdInfo,
+  TrackPathColorMode,
+} from "@/colorizer/types";
+
+// Standalone type for render canvas state params
+export type RenderCanvasStateParams = {
```

Comment:
Big change here. I didn't want `colorizer` to be importing from outside modules at all, so I ended up defining the fields of this type here. The selector for these fields was moved out of `colorizer` and into `state`.